CLI using cobra

areed opened the pull request:
Subcommands are implemented as methods on runners type, which holds
dependencies.
Integration tests with ginkgo run against an API.

emosbaugh commented on cli/cmd/channel_rm.go:
```diff
@@ -0,0 +1,35 @@
+var channelRmCmd = &cobra.Command{
+	Use:   "rm <id>",
+	Short: "Remove (archive) a channel",
+	Long:  `replicated channel rm 4d3d240ea1ec4dab0be3b2105ff4b4ed`,
```
lets remove long descriptions that echo command

emosbaugh commented on cli/cmd/channel_inspect.go:
```diff
@@ -0,0 +1,59 @@
+	channelCmd.AddCommand(channelInspectCmd)
+}
+
+func (r *runners) channelInspect(cmd *cobra.Command, args []string) error {
```
im concerned about how parseable this output is. a lot of ppl will use cli for automation

emosbaugh commented on cli/cmd/channel_create.go:
```diff
@@ -0,0 +1,34 @@
+		return err
+	}
+
+	return print.Channels(r.w, allChannels)
```
would be awesome if printers supported formats eg tabs, json, yaml

emosbaugh commented on cli/cmd/channel_inspect.go:
```diff
@@ -0,0 +1,59 @@
+var channelInspectCmd = &cobra.Command{
+	Use:   "inspect",
+	Short: "Show full details for a channel",
+	Long:  `replicated channel inspect be52315888f23408e2e4dc9242d4cc2c`,
```
lets remove long descriptions that echo command

areed commented:
I fixed the short/long/use help text. Still todo:
* docs  from godoc and generated like on-prem cli
* print yaml or json
* accept appID in addition to slug
* split channel inspect into multiple commands with more parseable output

emosbaugh commented:
@areed wanna create issues in github?

areed commented:
added
